packages/serverpod/lib/src/relic/routes/static_directory.dart

```diff
@@ -27,8 +27,23 @@ class RouteStaticDirectory extends Route {
   /// The path to the directory to serve static files from.
   final String? basePath;
 
+  /// A regular expression that will be used to determine if a path should not
+  /// be cached.
+  late final RegExp? noCachePathRegexp;
+
   /// Creates a static directory with the [serverDirectory] as its root.
-  RouteStaticDirectory({required this.serverDirectory, this.basePath});
+  /// If [basePath] is provided, the directory will be served from that path.
+  /// If [noCachePathPattern] is provided, paths matching the provided pattern
+  /// will not be cached. (Otherwise, all paths will be cached for a year.)
+  RouteStaticDirectory({
+    required this.serverDirectory,
+    this.basePath,
+    String? noCachePathPattern,
```

Comment:
Suggestion: Modify this to take the RegExp directly, I think this makes the interface more clear what we expect.

It may also be a good idea to take a list rather than a single value as that would make it easier to opt out of caching for different subroutes.

e.g.

```
[
  RegExp('images/*'),
  RegExp('hello.html'),
]

Reply:
Done